dptb/nn/energy.py

```
@@ -113,6 +123,178 @@ def forward(self,
+            solver.factorize(A)
+            
+            def matvec(b):
+                return solver.solve(A, b)
+                
+            Op = LinearOperator((N, N), matvec=matvec, dtype=A.dtype)
+            
+            try:
+                # Use larger NCV to help convergence, especially for clustered eigenvalues
+                ncv =  max(2*self.neig + 1, 20)
+                vals, vecs = eigsh(A=hk, M=M, k=self.neig, sigma=self.sigma, OPinv=Op, mode=self.mode, which="LM", ncv=ncv)
+            except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)